### ISSUE TITLE
Hubspot integration - Notice and consent / legitimate interest (GDPR) field can't be linked

### ISSUE THREAD

**Issue by carinameissl** [verbb/formie#821]
**Description**

I'm trying to connect our Formie form with a form in hubspot, everything works perfectly except the Notice and consent / legitimate interest (GDPR) field which is default on every form. (Screenshot 1)
This field is not shown in the integrations map-to-form settings in Formie unlike all the other fields. (Screenshot 2)
However, we need to link to this field so we can save newsletter signups for instance. Ist there a way to connect this field with Formie? 

<img width="753" alt="Screenshot 2022-04-06 at 14 42 59" src="https://user-images.githubusercontent.com/62565016/161976978-a42d5f79-e6fe-4981-b331-c855e34b70b8.png">

<img width="1130" alt="Screenshot 2022-04-06 at 16 17 58" src="https://user-images.githubusercontent.com/62565016/161996001-2187fd22-c667-46ff-b6cf-218327d17c78.png">

**Steps to reproduce**
1.
2.

### Form settings
Please pick from the options below.

- Multi-page form: Yes or No
- Submission Method: Ajax
- Client-side Validation: Yes or No
- Custom Form Templates: Yes

**Additional info**
- Plugin version: 1.5.4
- Craft version:  3.7.27.2
- Multi-site: Yes

**Additional context**

**Comment by engram-design** [comment]
Added for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.12",
  "...": "..."
}
```

Then run `composer update`.

**Comment by carinameissl** [comment]
Hi @engram-design,
thanks so much for the quick response and solution!

I’ve installed the new dev version and the field is available now and can be linked, however, I always get a 500 server error now when submitting the form (even when I turn the hubspot integration off) - also tried to disable Formie and enable it again. 
Do you have an idea what the issue could be? 

<img width="756" alt="Screenshot 2022-04-07 at 13 00 05" src="https://user-images.githubusercontent.com/62565016/162184554-69d59a67-c5a4-4e8d-a9ee-6d6708df6a55.png">


Thanks, 
Carina

**Comment by engram-design** [comment]
Hmm would you be able to look in your log file - storage/logs/formie.log and that should give you a bit more detail about the error?

I did certainly test things with and without the GDPR field, but you did mention you were on 1.5.4 which was a few versions back so it might be something unrelated.

**Comment by carinameissl** [comment]
When submitting I found this error in the log files (in the formie log files were no errors) #11 /vendor/verbb/formie/src/elements/Submission.php(444): yii\base\Model->validate(NULL, true)

**Comment by engram-design** [comment]
Thanks for that - was that the full error? Just trying to track it down, as it looks like that might've been the 11th step in the trace. But it looks like there was an issue validating the submission - but at the base element level, which a Submission inherits from.

**Comment by carinameissl** [comment]
I deleted the log folder and submitted the form again - here's the full stack trace, hope that helps: 

```
2022-04-08 09:20:04 [-][1][46743d81a46cfc26a63c7aa4589a61b7][error][yii\base\UnknownPropertyException] yii\base\UnknownPropertyException: Setting unknown property: craft\validators\SiteIdValidator::allowDisabled in /app/current/vendor/yiisoft/yii2/base/Component.php:209
Stack trace:
#0 /app/current/vendor/yiisoft/yii2/BaseYii.php(558): yii\base\Component->__set('allowDisabled', true)
#1 /app/current/vendor/yiisoft/yii2/base/BaseObject.php(107): yii\BaseYii::configure(Object(craft\validators\SiteIdValidator), Array)
#2 [internal function]: yii\base\BaseObject->__construct(Array)
#3 /app/current/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs(Array)
#4 /app/current/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build('craft\\validator...', Array, Array)
#5 /app/current/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('craft\\validator...', Array, Array)
#6 /app/current/vendor/yiisoft/yii2/validators/Validator.php(231): yii\BaseYii::createObject(Array)
#7 /app/current/vendor/yiisoft/yii2/base/Model.php(468): yii\validators\Validator::createValidator('craft\\validator...', Object(verbb\formie\elements\Submission), Array, Array)
#8 /app/current/vendor/yiisoft/yii2/base/Model.php(420): yii\base\Model->createValidators()
#9 /app/current/vendor/yiisoft/yii2/base/Model.php(189): yii\base\Model->getValidators()
#10 /app/current/vendor/yiisoft/yii2/base/Model.php(354): yii\base\Model->scenarios()
#11 /app/current/vendor/verbb/formie/src/elements/Submission.php(444): yii\base\Model->validate(NULL, true)
#12 /app/current/vendor/verbb/formie/src/controllers/SubmissionsController.php(488): verbb\formie\elements\Submission->validate()
#13 [internal function]: verbb\formie\controllers\SubmissionsController->actionSubmit()
#14 /app/current/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#15 /app/current/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#16 /app/current/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('submit', Array)
#17 /app/current/vendor/craftcms/cms/src/web/Application.php(287): yii\base\Module->runAction('formie/submissi...', Array)
#18 /app/current/vendor/craftcms/cms/src/web/Application.php(596): craft\web\Application->runAction('formie/submissi...', Array)
#19 /app/current/vendor/craftcms/cms/src/web/Application.php(266): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#20 /app/current/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#21 /app/current/web/index.php(22): yii\base\Application->run()
#22 {main}
2022-04-08 09:20:04 [-][1][46743d81a46cfc26a63c7aa4589a61b7][info][application] $_GET = []
```

**Comment by engram-design** [comment]
Ah, looks like I didn't realise `allowDisabled` was only added in Craft 3.7.32, so it was indeed unrelated. That should be fixed in the latest branch (just run that command again)

**Comment by engram-design** [comment]
Fixed in [1.5.13](https://github.com/verbb/formie/releases/tag/1.5.13)

**Comment by carinameissl** [comment]
@engram-design Solved all of the issues and works perfectly fine now, thanks a lot!
